internal/testrunner/runners/system/runner.go

```diff
@@ -1150,6 +1151,9 @@ func (r *runner) prepareScenario(ctx context.Context, config *testConfig, svcInf
 		}
 	}
 	r.resetAgentLogLevelHandler = func(ctx context.Context) error {
+		if r.options.RunTestsOnly {
+			return nil
+		}
```

Comment:
Nit. In these cases, could we directly avoid adding the handler?
```suggestion
	if r.options.RunTestsOnly {
		r.resetAgentLogLevelHandler = func(ctx context.Context) error {
```

Reply:
It could be done, but there are handlers that have more conditions to not run. For instance 

```
	r.removeAgentHandler = func(ctx context.Context) error {
		if r.runTestsOnly {
			return nil
		}
		// When not using independent agents, service deployers like kubernetes or custom agents create new Elastic Agent
		if !r.runIndependentElasticAgent && !svcInfo.Agent.Independent {
			return nil
		}
```

I think I'll keep all the handlers as they are to keep the same pattern in all of them.